apps/arweave/src/ar_mining_worker.erl

```diff
@@ -85,24 +117,17 @@ computed_hash(Worker, computed_h2, H2, Preimage, Candidate) ->
 %% ordering. The previous block is chosen only after the mining solution is found (if
 %% we choose it in advance we may miss a better option arriving in the process).
 %% Also, a mining session may (in practice, almost always will) span several blocks.
+-spec set_difficulty(Worker :: pid(), DiffPair :: {non_neg_integer(), non_neg_integer()}) -> ok.
 set_difficulty(Worker, DiffPair) ->
-	gen_server:cast(Worker, {set_difficulty, DiffPair}).
-
-set_cache_limits(Worker, SubChunkCacheLimit, VDFQueueLimit) ->
-	gen_server:cast(Worker, {set_cache_limits, SubChunkCacheLimit, VDFQueueLimit}).
+	gen_server:cast(Worker, ?MSG_SET_DIFFICULTY(DiffPair)).
 
-%% @doc Returns true if the mining candidate belongs to a valid mining session. Always assume
-%% that a coordinated mining candidate is valid (its cm_lead_peer is set)
-is_session_valid(_State, #mining_candidate{ cm_lead_peer = Peer })
-		when Peer /= not_set ->
-	true;
-is_session_valid(
-		#state{ active_sessions = Sessions },
-		#mining_candidate{ session_key = SessionKey }) ->
-	sets:is_element(SessionKey, Sessions).
+-spec set_cache_limits(Worker :: pid(), SubchunkCacheLimitBytes :: non_neg_integer(), VDFQueueLimit :: non_neg_integer()) -> ok.
```

Comment:
```suggestion
-spec set_cache_limits(Worker :: pid(), SubChunkCacheLimitBytes :: non_neg_integer(), VDFQueueLimit :: non_neg_integer()) -> ok.
```

For consistency with the rest of the code, I think we should do `SubChunk` rather than `Subchunk` (there are few places to update, but I'll just comment here)

Reply:
Or CacheLimitBytes as you suggest below: https://github.com/ArweaveTeam/arweave/pull/697/files#r1966570633